pylint/checkers/typecheck.py

```diff
@@ -1496,6 +1500,46 @@ def visit_call(self, node: nodes.Call) -> None:
             ):
                 self.add_message("missing-kwoa", node=node, args=(name, callable_name))
 
+    @staticmethod
+    def _keyword_argument_is_in_all_decorator_returns(
+        func: nodes.FunctionDef, keyword: str
+    ) -> bool:
+        """Check if the keyword argument exists in all signatures of the
+        return values of all decorators of the function.
+        """
+        if not func.decorators:
+            return False
+
+        for decorator in func.decorators.nodes:
+            inferred = safe_infer(decorator)
+
+            # If we can't infer the decorator we assume it satisfies consumes
+            # the keyword, so we don't raise false positives
+            if not inferred:
```

Comment:
Not really related to this MR but if we have to handle confidence level, this kind of return where we do not exactly know if inference was used or not in the calling code will be a headache later. Makes me thing that implementing confidence properly will be an enormous work, maybe not even worth it.

Reply:
Thankfully we have `UNDEFINED` 😄